backend/database.js

```diff
@@ -36,32 +37,49 @@ if (!fs.existsSync(dataFolder)) {
   fs.mkdirSync(dataFolder, { mode: 0o750 })
 }
 
+// Streams stored contract log entries since the given entry hash (inclusive!), most recent first.
 sbp('sbp/selectors/register', {
   'backend/db/streamEntriesAfter': async function (contractID: string, hash: string): Promise<*> {
     const latestHEADinfo = await sbp('chelonia/db/latestHEADinfo', contractID)
     if (!latestHEADinfo) {
       throw Boom.notFound(`contractID ${contractID} doesn't exist!`)
     }
-    let { HEAD: currentHEAD } = latestHEADinfo
+    const entries = []
+    // Number of entries pushed.
+    let counter = 0
+    let currentHash = hash
     let prefix = '['
+    try {
+      while (currentHash && (entries.length < MAX_EVENTS_AFTER)) {
+        const entry = await sbp('chelonia/db/getEntry', currentHash)
+        if (entry) {
+          entries.push(entry)
+          currentHash = await sbp('chelonia/db/get', `next=${currentHash}`)
+        }
+      }
```

Comment:
The entire idea behind streams is that they allow us to avoid having to read all of the data into memory.

With the way this is currently implemented, we are reading all of the entries into memory, and therefore there's no reason to even return a readable stream, we could just return the array we read into memory.

But we do want to return a readable stream. What if there's a contract with 1000000 entries in it? Reading all of those entries into memory and returning them would be very hard on the server, and could crash it.

So, could you please remove this code, and instead use the readable stream below to pull out entries on an as-needed basis (e.g. no more than 5 entries per call to `read()`)